library/modules/Gui.cpp

```diff
@@ -934,12 +946,63 @@ df::unit *Gui::getAnyUnit(df::viewscreen *top)
         return NULL;
     }
 
+    if (VIRTUAL_CAST_VAR(screen, df::viewscreen_reportlistst, top))
+        return vector_get(screen->units, screen->cursor);
+
+    if (VIRTUAL_CAST_VAR(screen, df::viewscreen_announcelistst, top))
+    {
+        auto *report = vector_get(screen->reports, screen->sel_idx);
```

Comment:
Should check for null. I don't think that should ever happen with this screen, but it could with raid reports, or it might change in the future. Also, `vector_get` is designed to return null instead of crashing (although admittedly that covers bad indices, nothing specific to vectors containing null pointers), so you might as well check for it.